compliance/flexera/spot_security_report/flexera_spot_security_report.pt

```diff
@@ -0,0 +1,381 @@
+  label "Allow/Deny Spot Accounts List"
+  description "A list of allowed or denied Spot Accounts"
+  allowed_pattern /^([a-zA-Z-_]+-[a-zA-Z0-9-_]+-[0-9-_]+,*|)+$/
+  default []
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+credentials "auth_spotinst" do
+  schemes "api_key"
+  label "SpotInst"
+  description "Select the SpotInst Credential from the list."
+  tags "provider=spotinst"
+end
+
+
+###############################################################################
+# Pagination
+###############################################################################
+
+pagination "pagination_spot" do
+  get_page_marker do
+    body_path jq(response, ".response.paginationInfo.nextKey")
+  end
+  set_page_marker do
+    query "paginationKey"
+  end
+end
+
+###############################################################################
+# Datasources & Scripts
+###############################################################################
+
+# https://api.spotinst.io/setup/account
+# Docs: https://docs.spot.io/api/#tag/Accounts/operation/OrganizationsAndAccountsGetAccounts
+datasource "ds_spotinst_accounts" do
+  request do
+    auth $auth_spotinst
+    host "api.spotinst.io"
+    path "/setup/account"
+  end
+  result do
+    encoding "json"
+    collect jq(response, ".response.items[]") do
+      # response field
+      field "account", jq(col_item, ".")
+    end
+  end
+end
+
+datasource "ds_spotinst_accounts_filtered" do
+  run_script $js_azure_subscriptions_filtered, $ds_spotinst_accounts, $param_spotinst_accounts_allow_or_deny, $param_spotinst_accounts_list
+end
+
+script "js_azure_subscriptions_filtered", type: "javascript" do
+  parameters "ds_spotinst_accounts", "param_spotinst_accounts_allow_or_deny", "param_spotinst_accounts_list"
+  result "result"
+  code <<-EOS
+  if (param_spotinst_accounts_list.length > 0) {
+    result = _.filter(ds_spotinst_accounts, function(subscription) {
+      include = _.contains(param_spotinst_accounts_list, subscription['account']['accountId']) || _.contains(param_spotinst_accounts_list, subscription['account']['name'])
+
+      if (param_spotinst_accounts_allow_or_deny == "Deny") {
+        include = include
+      }
+
+      return include
+    })
+  } else {
+    result = ds_spotinst_accounts
+  }
+EOS
+end
+
+datasource "ds_spot_compliances" do
+  # iterate $ds_spotinst_accounts_filtered
+  request do
+    auth $auth_spotinst
+    host "api.spotinst.io"
+    path "/security/v1/compliances"
+    # header "Content-Type", "application/json"
+    # query "accountId", val(val(iter_item,"account"), "accountId")
+  end
+  result do
+    encoding "json"
+    collect jq(response, ".response.items[]") do
+      # iteration fields
+      # field "account", jq(iter_item, ".account")
+      # response field
+      field "compliance", jq(col_item, ".")
+    end
+  end
+end
```

Comment:
This datasource does not seem to be used anywhere. Was this supposed to be used as part of the `ds_security_report` datasource, or should it be removed?

Reply:
This gets a list of "compliances" -- which are compliance standards organizations can measure against.. idea was to potentially use that to enhance the report (or create new incident(s)) and showing % compliance for each of those compliance standards like the Spot dashboard does but I did not implement that yet